$NetBSD: patch-ba,v 1.1 2007/12/01 12:28:04 rillig Exp $

On NetBSD, the sigcontext struct is so well-protected from userland that
I didn't find a way to access it.

--- src/EnginePl/machine.c.orig	2007-01-04 11:35:13.000000000 +0100
+++ src/EnginePl/machine.c	2007-12-01 13:18:32.000000000 +0100
@@ -469,7 +469,7 @@ SIGSEGV_Handler(int sig, struct sigconte
 static void
 SIGSEGV_Handler(int sig, siginfo_t * si)
 
-#elif defined(M_ix86_bsd) || defined(M_powerpc_bsd) || defined(M_sparc_bsd)
+#elif (defined(M_ix86_bsd) || defined(M_powerpc_bsd) || defined(M_sparc_bsd)) && !defined(__NetBSD__)
 static void
 SIGSEGV_Handler(int sig, int code, struct sigcontext *scp)
 
@@ -517,7 +517,7 @@ SIGSEGV_Handler(int sig)
 
   WamWord *addr = (WamWord *) si->si_addr;
 
-#elif defined(M_ix86_bsd)
+#elif defined(M_ix86_bsd) && !defined(__NetBSD__)
 
   WamWord *addr = (WamWord *) scp->sc_err;
 
